### PR TITLE
Test class factory

### PR DESCRIPTION
**Pull request by wassoares**
- Added more tests to coverage of the ClassFactory class, and created a stub class to assist in these tests;

### CODE REVIEW

File: src/test/java/cloud/fogbow/common/util/ClassFactoryTest.java [hunk 1]
```diff
@@ -1,57 +1,149 @@
...
+				parameter2);
+
+		// verify
+		Assert.assertNotNull(object);
+	}
+
+	// test case: When invoking the createPluginInstance method without parameters
+	// and without a valid class path, it must throw a FatalErrorException.
+	@Test(expected = FatalErrorException.class) // veriry
+	public void testCreatePluginInstanceNoParametersAndInvalidClassPath() {
```
[the remark this answers is not in the document]
Comment (thiagomanel) [reply src/test/java/cloud/fogbow/common/util/ClassFactoryTest.java]: ignore this comment. Now, I realized the codebase follow this pattern.

File: src/test/java/cloud/fogbow/common/util/ClassFactoryTest.java [hunk 2]
```diff
@@ -1,57 +1,149 @@
...
-    public void createPluginInstanceWithNonExistingClass() {
-        String invalidClassPath = "invalid.class.path.InvalidClass";
-        classFactory.createPluginInstance(invalidClassPath);
-    }
+	private static final String ONE_PARAMETER = "parameter1";
+	private static final String INVALID_CLASS_PATH = "invalid.class.path.InvalidClass";
+	private static final String TWO_PARAMETER = "parameter2";
+	private static final String STUB_CLASS_PATH = "cloud.fogbow.common.stubs.StubClassForFactory";
+	private static final String STUB_CLASS_WITH_PRIVATE_CONSTRUCTOR_PATH = "cloud.fogbow.common.stubs.StubClassForFactoryWithPrivateConstructor";
+
```
[the remark this answers is not in the document]
Comment (wassoares) [reply src/test/java/cloud/fogbow/common/util/ClassFactoryTest.java]: Yes, it is possible to add this class as an inner class in ClassFactoryTest. I will do this.

File: src/test/java/cloud/fogbow/common/util/ClassFactoryTest.java [hunk 3]
```diff
@@ -1,57 +1,149 @@
...
+	@Before
+	public void setUp() {
+		this.classFactory = new ClassFactory();
+	}
+
+	// test case: When invoking the createPluginInstance method without parameters
+	// and with a valid class path, it must create the object from the constructor
+	// without parameters of the specified class.
+	@Test
```
[the remark this answers is not in the document]
Comment (wassoares) [reply src/test/java/cloud/fogbow/common/util/ClassFactoryTest.java]: I'll fix this.

File: src/test/java/cloud/fogbow/common/util/ClassFactoryTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (thiagomanel) [reply src/test/java/cloud/fogbow/common/util/ClassFactoryTest.java]: @wassoares you missed this one

[the remark this answers is not in the document]
Comment (wassoares) [reply src/test/java/cloud/fogbow/common/util/ClassFactoryTest.java]: I updated the tests and now it is no longer necessary to use a stub class with a private constructor.

**Comment by wassoares** [comment]
About the general comment...
"Would not be simple to have an interface like StubClassForFactory (String ... args) instead of three different methods" 

Seemingly the getConstructor method of the java reflect API, can not match a constructor in which the number of parameters used is not well specified, preventing the construction for the example "StubClassForFactory (String ... args ) ".

**Pull request merged**